### PR TITLE
`Development`: Add server tests for practice participations in student course dashboard

### PR DESCRIPTION
**Pull request by EneaGore**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

In `ExerciseMetricsRepository` `findLatestSubmissionDatesForUser` and `findLatestSubmissionDates` we are filtering out the practice participations using `testRun = False`. We have to write server tests for this functionality.

### Description

Adds two tests for the above stated methods. 

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2

## Summary by CodeRabbit

- **Tests**
  - Added new tests to validate the latest submission dates for exercises.
  - Enhanced test coverage for retrieving submission dates by individual users.

### CODE REVIEW

**Review by JohannesStoehr — DISMISSED** [review]

File: src/test/java/de/tum/in/www1/artemis/MetricsIntegrationTest.java [hunk 1]
```diff
@@ -147,5 +154,44 @@ void shouldReturnLatestSubmission() throws Exception {
 
             assertThat(latestSubmissions).isEqualTo(expectedMap);
         }
+
+        @Test
+        @WithMockUser(username = STUDENT_OF_COURSE, roles = "USER")
+        void shouldFindLatestSubmissionDates() throws Exception {
+            Set<Long> exerciseIds = new HashSet<Long>();
+            final var exercises = exerciseRepository.findAllExercisesByCourseId(course.getId()).stream()
```
[the remark this answers is not in the document]
Comment (EneaGore) [reply src/test/java/de/tum/in/www1/artemis/MetricsIntegrationTest.java]: You are right, i added a new util method to have at least one practice Participation and 2 different users.

**Review by edkaya — APPROVED** [review]
Corrosponding test cases pass 👍

**Review by marlon-luca-bu — APPROVED** [review]
Code

**Review by egekurt123 — APPROVED** [review]
Tests pass locally

**Pull request merged**